### ISSUE TITLE
Is there a way to make my kolbot pick up gold in town?

### ISSUE THREAD

**Issue by SirYac** [kolton/d2bot-with-kolbot#1201]
I don't want my leader to do gambling (it destroys the game sometimes, when gambling takes too long) so basically it always has maximum gold in stash and with it.
When selling items to NPC the excess of gold just falls on the ground in massive numbers of 35k piles.

Is there any way i can make my Followers (which do gamble) pick up that gold in town?

Thank you!

**Comment by 5noop** [comment]
As a "dirty fix" you can add `Pickit.pickItems();` at the end of the doChores function in Town.js, just before `return true;`, then your follower will pick the gold if 1) it's in his pickit 2) if he runs doChores after your leader and in the same act. It won't always work but it's easy and probably better than nothing.

If you want a better way you have to write a script that either make your follower go the each NPC at the end of the game to pick gold (quite easy, not very efficient), or you write a script to run at the game start with communication between your leader and follower to drop/pick gold when needed. Gambling.js does something like that but it would need to be adapted.

**Comment by SirYac** [comment]
`	// Do town chores
	doChores: function () {
		if (!me.inTown) {
			this.goToTown();
		}

		var i,
			cancelFlags = [0x01, 0x02, 0x04, 0x08, 0x14, 0x16, 0x0c, 0x0f, 0x19, 0x1a];

		if (me.classid === 4 && Config.FindItem && Config.FindItemSwitch) { // weapon switch fix in case last game dropped with item find switch on
			Precast.weaponSwitch(Math.abs(Config.FindItemSwitch - 1));
		}

		if (Config.MFSwitchPercent) {
			Precast.weaponSwitch(Math.abs(Config.MFSwitch - 1));
		}

		if (Precast.haveCTA > -1) {
			Precast.weaponSwitch(Math.abs(Precast.haveCTA - 1));
		}

		this.heal();
		this.identify();
		this.shopItems();
		this.fillTome(518);

		if (Config.FieldID) {
			this.fillTome(519);
		}

		this.buyPotions();
		this.clearInventory();
		Item.autoEquip();
		this.buyKeys();
		this.repair();
		this.gamble();
		this.reviveMerc();
		Cubing.doCubing();
		Runewords.makeRunewords();
		this.stash(true);
		this.clearScrolls();

		for (i = 0; i < cancelFlags.length; i += 1) {
			if (getUIFlag(cancelFlags[i])) {
				delay(500);
				me.cancel();

				break;
			}
		}

		me.cancel();
		
		Pickit.pickItems();
		
		return true;
	},

	checkQuestItems: function () {`

Added what you suggested but it doesn't work, also i have gold in my pickit. What did i to wrong?

**Comment by 5noop** [comment]
Hm yeah it won't work if your bot stash stuff for instance, try to put those 2 lines after `this.buyKeys();`:
```
me.cancel();
Pickit.pickItems();
```

**Comment by aim2kill** [comment]
Autosmurf do something like that to. Share gold between them

**Comment by DarkHorseDre** [comment]
Also seems https://github.com/kolton/d2bot-with-kolbot/wiki/Gambling-System does this
gambling.js drops gold
Gamble.js picks it - seems legit?

**Comment by AlexyGalaxy** [comment]
Simple fix is to add Pickit.pickItems(); anywhere in doChores() in town.js 
I recommend putting it before this.buyPotions(); so that if items are picked up they can also be sold for gold. Also, when you join a game and you see gold on the floor in town, you should pick it up as you walk by.



```
	doChores: function (repair = false) {
		if (!me.inTown) {
			this.goToTown();
		}

		var i,
			cancelFlags = [0x01, 0x02, 0x04, 0x08, 0x14, 0x16, 0x0c, 0x0f, 0x19, 0x1a];

		Attack.weaponSwitch(Attack.getPrimarySlot());

		this.heal();
		this.identify();
		this.shopItems();
		this.fillTome(518);

		if (Config.FieldID) {
			this.fillTome(519);
		}

		print("Trying to pick up Items");
		Pickit.pickItems();
		this.buyPotions();
		this.clearInventory();
		Item.autoEquip();
		this.buyKeys();
		this.repair(repair);
		this.gamble();
		this.reviveMerc();
		Cubing.doCubing();
		Runewords.makeRunewords();
		this.stash(true);
		this.clearScrolls();

		for (i = 0; i < cancelFlags.length; i += 1) {
			if (getUIFlag(cancelFlags[i])) {
				delay(500);
				me.cancel();

				break;
			}
		}

		me.cancel();

		return true;
	},
```

**Comment by attovide** [comment]
Cool work

**Comment by DarkHorseDre** [comment]
Be careful adding pickit intown for all items and gold - IIRC if there are items/gold there intentionally (e.g. for the gamble bot) then you will interfere with that

**Comment by AlexyGalaxy** [comment]
I dont think I ever used the gamble bot, I always set my bots to gamble if they hit a unique max number of gold and will continue to gamble until their max number of gold reaches a unique low number. If you are joining public matches I suggest adding the lines of code i provided as many bots you join will drop gold at shops once they reach max gold.

**Comment by DarkHorseDre** [comment]
It was an example, and aimed at all users.
don't forget the implications of selling any items found on the ground

**Comment by AlexyGalaxy** [comment]
I disagree with you on that. If you have your game set to public anything you drop is fair game. This is classic D2, if you want privacy then make a private game.

**Comment by DarkHorseDre** [comment]
> I disagree with you on that. If you have your game set to public anything you drop is fair game. This is classic D2, if you want privacy then make a private game.

I wasn't speaking of public games and other peoples items.  nevermind, I wasn't aiming my response at you anyway, but to people who will read this thread.
